### ISSUE TITLE
Add action to "Batch Compound Clip"

### ISSUE THREAD

**Issue by kltdwrds** [fcpxhacks/fcpxhacks#3318]
Hello! I am a software engineer and surf videographer.  I use Compound Clips to batch export timeline clips as individual files with effects applied.  I tried CommandPost's Batch Export as an alternative to compound clips, and while it works perfectly as advertised, thought it's not ideal giving up control of my laptop for duration of automated tasks.

Idea: automate batch creation of compound clips (via timeline's selected clips), and use native [batch sharing](https://support.apple.com/guide/final-cut-pro/batch-sharing-verd4e0cf686/mac) to export the clips.  I estimate this approach to be faster than Batch Export Plugin, and users will retain control during the actual background rendering period.  

Open to feedback on this idea, and possibly guidance on best approach to implement cc @latenitefilms 

Cheers and mahalo for the great project!  I visited fcp.cafe and immediately learned some useful things 🤙 🏄

**Comment by kltdwrds** [comment]
Any thoughts from the commandpost team on the above?  Thanks again for the awesome project.

**Comment by latenitefilms** [comment]
What about if you "Send to Compressor" using Timeline Batch Export?

**Comment by latenitefilms** [comment]
Alternatively, in theory, you could just duplicate the CommandPost Timeline Batch Export plugin and instead of triggering a Share Menu Item you could just trigger a create Compound Clip.

**Comment by randomeizer** [comment]
We could add that pretty easily @latenitefilms? Just call it "Batch Compound Clip" or something?

**Comment by latenitefilms** [comment]
Absolutely, but I feel like Timeline Batch Export needs a complete redesign anyway (or better yet, spun off into it's own app), as it's mostly my code, and doesn't use `Rx`. I'd imagine if we duplicated for Compound Clips, we'd get the same amount of bug reports as we do for Timeline Batch Export?

**Comment by kltdwrds** [comment]
> What about if you "Send to Compressor" using Timeline Batch Export?

I gave this a try and it's pretty close to the solution I'm looking for -- noticed one pain point where the project name is used as the filename, which results in an export error.  I could have manually renamed the files in compressor, but my compound clips approach would be preferred in that case.

<img width="373" alt="Screenshot 2024-01-29 at 4 39 27 PM" src="https://github.com/CommandPost/CommandPost/assets/5615037/e97bea3d-b719-4733-856e-7b90b1701c6b">
<img width="847" alt="Screenshot 2024-01-29 at 4 40 45 PM" src="https://github.com/CommandPost/CommandPost/assets/5615037/6b3af1ff-8645-4a3e-acce-fd6e57b12539">
<img width="501" alt="Screenshot 2024-01-29 at 4 43 41 PM" src="https://github.com/CommandPost/CommandPost/assets/5615037/0b401de6-40e6-48d2-8ae1-e60521f1b185">

**Comment by latenitefilms** [comment]
Ah yes, I don't think there's anyway we can control that.

Will look into a "Batch Compound Clip" action.

**Comment by skymakai** [comment]
An issue I've found with batch exporting to Compressor is this: Compressor loads all the clips at once and into memory, limiting the amount of clips to the amount of memory available.
